File: Xamarin.Android Client/RecommendationSystemsClient/Fragments/BaseFragment.cs
using Android.OS;
using Android.Views;
using CheeseBind;

namespace RecommendationSystemsClient.Fragments
{
    public abstract class BaseFragment : Android.Support.V4.App.Fragment
    {
        protected abstract int LayoutResource { get; }

        public override View OnCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState)
        {
            var root = inflater.Inflate(LayoutResource, container, false);

            /* Bind CheeseKnife */
            Cheeseknife.Bind(this, root);

            return root;
        }
    }
}

File: Native Android Client/app/src/main/java/com/distributedsystems/recommendationsystemclient/Fragments/BaseFragment.java
package com.distributedsystems.recommendationsystemclient.Fragments;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import butterknife.ButterKnife;

public abstract class BaseFragment extends Fragment{
    public String[] poiCategoriesAvailable = new String[]{
        "Arts & Entertainment",
        "Bars",
        "Food",
        "Unknown Category"
    };

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater,
                             @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {

        View root = inflater.inflate(getResourceLayout(), container, false);

        /* Bind ButterKnife */
        ButterKnife.bind(this, root);

        return root;
    }

    public abstract int getResourceLayout();
}
